tests/test_calc_error.py

```diff
@@ -0,0 +1,382 @@
+            m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt = 1)
+        self.assertEqual(
+            "Model unstable- NAN reached in simulation (t=1800.0) before cutoff threshold. Cutoff threshold is 1900.0, or roughly 95.0% of the data",
+            str(cm.exception)
+        )
+        
+        # Checks to see if m1 throws the same exception. 
+        with self.assertRaises(ValueError):
+            m1.calc_error(m1_sim_results.times, m1_sim_results.inputs, m1_sim_results.outputs, dt = 1)
+        self.assertEqual(
+            "Model unstable- NAN reached in simulation (t=1800.0) before cutoff threshold. Cutoff threshold is 1900.0, or roughly 95.0% of the data",
+            str(cm.exception)
+        )
+
+        # Checks to see if stability_tolerance throws Warning rather than an Error when the model goes unstable after threshold
+        with self.assertWarns(UserWarning) as cm:
+            m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, 
+                     dt = 1, stability_tol=0.7)
+        self.assertEqual(
+            'Model unstable- NaN reached in simulation (t=1800.0)',
+            str(cm.warning)
+        )
+
+    def test_MSE(self):
+        """
+        Base tests that ensure first-level workability within calc_error
+        """
+        m = ThrownObject()
+        m2 = ThrownObject()
+        results = m.simulate_to_threshold(save_freq=0.5)
+        gt = m.parameters.copy()
+
+        # Arbitrary Test to ensure that both models are behaving the same way
+        self.assertEqual(m.calc_error(results.times, results.inputs, results.outputs),
+                            m2.calc_error(results.times, results.inputs, results.outputs))
+        
+        resultsm2 = m2.simulate_to_threshold(save_freq = 0.5)
+        m2.parameters['throwing_speed'] = 35
+        key = ['throwing_speed']
+
+        previous = m2.calc_error(resultsm2.times, resultsm2.inputs, resultsm2.outputs)
+
+        self.assertNotEqual(m.calc_error(results.times, results.inputs, results.outputs),
+                            previous)
+        
+        m2.estimate_params(times = resultsm2.times, inputs = resultsm2.inputs, outputs = resultsm2.outputs, keys = key, dt = 1)
+
+        for i in key:
+            # We can compare with gt because m and m2 originally were the same, and gt was a copy of m.
+            self.assertAlmostEqual(m2.parameters[i], gt[i], 2)
+
+        self.assertNotEqual(m.calc_error(results.times, results.inputs, results.outputs),
+                    previous)
+        
+        self.assertLess(m2.calc_error(resultsm2.times, resultsm2.inputs, resultsm2.outputs), previous)
+
+        # By changing the 
+        for i in np.arange(0.1, 1, 0.1):
+            m.calc_error(results.times, results.inputs, results.outputs, dt=i)
+
+        for i in range(2, 10):
+            m.calc_error(results.times, results.inputs, results.outputs, dt=i)
```

Comment:
Can we add a comment on the purpose of these tests? It seems like they are just running the calc_error function for different dt values, but not actually saving or comparing the error results. Do we want to add a comparison of some sort?

Reply:
Yeah, the whole purpose of this was to ensure that given some base dt values, our calc_error method would return some value.

In terms of adding some comparisons, should I compare the errors between calc_error calls for a small and large dt value?

Reply:
I like the idea of testing that we can input different dt values, good test idea. 

Yes, I would try comparing the errors between two (or more) dt values. Once you've added a few of these checks, I don't think it's necessary to loop over quite so many dt values. Ultimately each iteration in these two loops are doing the same thing, so I think just a couple checks would be sufficient (but correct me if I'm misunderstanding)

Reply:
For the following code segment:
        small = m.calc_error(results.times, results.inputs, results.outputs, dt = 0.001)
        large = m.calc_error(results.times, results.inputs, results.outputs, dt = 0.1
This is true:
         self.assertGreater(small, large)


I was going to write a short description regarding why the larger dt value would result in a lower error, however, wouldn't this behavior be something that is dependent on the specific model we have simulated and the simulated results?

Reply:
Hm, I would have expected the opposite actually, that smaller dt gives smaller error. Let me pull your code and take a look

Reply:
I don't see any changes to this code

Reply:
Sorry, could you check again. I forgot to push the changes :(

Reply:
Sounds good to me.